qcodes/dataset/sqlite_base.py

```diff
@@ -382,8 +410,109 @@ def perform_db_upgrade_1_to_2(conn: SomeConnection) -> None:
+
+            idps_query = f"""
+                        SELECT independent
+                        FROM dependencies
+                        WHERE independent in {layout_ids}
+                        """
+            c = transaction(conn, idps_query)
+            independents = [l[0] for l in many_many(c, 'independent')]
+            dependents = set(layout_ids).difference(set(independents))
+
+            layout_query_r = """
+                            SELECT parameter, label, unit, inferred_from
+                            FROM layouts
+                            WHERE layout_id == {}
+                            """
+
+            for layout_id in layout_ids:
+                layout_query = layout_query_r.format(layout_id)
+                c = transaction(conn, layout_query)
+                res = many(c, 'parameter', 'label', 'unit', 'inferred_from')
+                (name, label, unit, inferred_from) = res
+                # get the data type
+                sql = f'PRAGMA TABLE_INFO("{result_table_name}")'
+                c = transaction(conn, sql)
+                for row in c.fetchall():
+                    if row['name'] == name:
+                        paramtype = row['type']
+                        break
+
+                if layout_id in independents:
+                    paramspec = ParamSpec(name=name, paramtype=paramtype,
+                                          label=label, unit=unit,
+                                          inferred_from=inferred_from)
+                    paramspecs[layout_id] = paramspec
+
+                elif layout_id in dependents:
+
+                    sp_query = f"""
+                            SELECT independent
+                            FROM dependencies
+                            WHERE dependent == {layout_id}
+                            ORDER BY axis_num ASC
+                            """
+                    c = transaction(conn, sp_query)
+                    setpoints = [l[0] for l in many_many(c, 'independent')]
+                    depends_on = [paramspecs[idp].name for idp in setpoints]
+
+                    paramspec = ParamSpec(name=name,
+                                          paramtype=paramtype,
+                                          label=label, unit=unit,
+                                          depends_on=depends_on,
+                                          inferred_from=inferred_from)
+                    paramspecs[layout_id] = paramspec
+
+            interdeps = InterDependencies(*paramspecs.values())
+            desc = RunDescriber(interdeps=interdeps)
+            yaml_str = desc.to_yaml()
+
+            sql = f"""
+                   UPDATE runs
+                   SET run_description = ?
+                   WHERE run_id == ?
+                   """
+            cur = conn.cursor()
+            cur.execute(sql, (yaml_str, run_id))
+
+            log.info(f"Upgrade in transition, run number {run_id}: OK")
```

Comment:
Do we have a benchmark on the time required to do  an actual upgrade of a "big" database. I am slightly concerned that this will be slow and users will get impatient and interrupt the process